src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,123 @@
+    private Node<K, V> [] table;
+    private int size;
+
+    public MyHashMap() {
+        if (table == null) {
+            table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = hash(key);
+        if (table[index] != null) {
+            if (equalsKey(table[index].key, key)) {
+                table[index].value = value;
+                return;
+            }
+            if (hasNext(table[index])) {
+                Node<K, V> current = table[index];
+                while (hasNext(current)) {
+                    current = current.next;
+                    if (equalsKey(current.key, key)) {
+                        current.value = value;
+                        return;
+                    }
+                    if (!hasNext(current)) {
+                        current.next = new Node<>(key, value, null);
+                        size++;
+                        return;
+                    }
+                }
+            } else {
+                table[index].next = new Node<>(key, value, null);
+                size++;
+            }
+        } else {
+            table[index] = new Node<>(key, value, null);
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> node : table) {
+            if (node == null) {
+                continue;
+            } else if (hasNext(node)) {
+                while (hasNext(node)) {
+                    if (equalsKey(node.key, key)) {
+                        return node.value;
+                    }
+                    node = node.next;
+                }
+            }
+            if (equalsKey(node.key, key)) {
+                return node.value;
+            }
+        }
         return null;
     }
 
+    private boolean hasNext(Node<K, V> node) {
+        return node.next != null;
+    }
+
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private boolean equalsKey(K firstKey, K secondKey) {
+        return firstKey == secondKey || firstKey != null && firstKey.equals(secondKey);
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : (Math.abs(key.hashCode()) % table.length);
+    }
+
+    private void resize() {
+        Node<K, V>[] newArray = (Node<K, V>[]) new Node[table.length << 1];
+        int oldCap = table.length;
+        int thresold = (int) (oldCap * LOAD_FACTOR);
+        if (size + 1 > thresold) {
+            fillNewArray(newArray);
+        }
+    }
+
+    private void fillNewArray(Node<K, V>[] newArray) {
```

Comment:
You can simplify a bit your code by moving this logic into the resize method and **reusing your put method** in order to locate your nodes in the new table.